dali/kernels/imgproc/roi.h

```diff
@@ -129,6 +130,38 @@ TensorListShape<ndims> ShapeFromRoi(span<const Roi<spatial_dims>> rois, int ncha
 }
 
 
+/**
+ * Overload for the case, when the memory layout is *HW,
+ * as the 1-channel image might be expressed in such layout.
+ */
+template <int ndims>
+TensorShape<ndims> ShapeFromRoi(const Roi<ndims> &roi) {
+  DALI_ENFORCE(all_coords(roi.hi >= roi.lo), "Cannot create a TensorShape from an invalid Roi");
+  TensorShape<ndims> ret;
+  auto e = roi.extent();
+  auto ridx = ndims;
+  for (size_t idx = 0; idx < ndims; idx++) {
+    ret[--ridx] = e[idx];
+  }
+  return ret;
```

Comment:
You want to reverse the order here, right?
```suggestion
  std::reverse(e);
  return TensorShape<ndims>(e.begin(), e.end());
```
this should work, I guess

Reply:
maybe better:
```c++
TensorShape<ndim> ret;
std::reverse_copy(e.begin(), e.end(), ret.begin());
return ret;
```

Reply:
Good point. Followed up in https://github.com/NVIDIA/DALI/pull/4375